File: new.py
import math
import sys


def maxprofit(N, K, A, B):

    A.sort()
    i=0
    j=1
    profit = 0
    maxp = 0
    while(i<=j):
        if(j==N):
            i+=1
            j=i+1
        if(A[i] + A[j] <= K and j<N):
            profit += B[i] + B[j]
            j+=1
        else:
            break

        maxp = max(maxp, profit)
        profit = 0
        
    return maxp




def main():

    N = int(sys.stdin.readline().strip())

    K = int(sys.stdin.readline().strip())

    A = []
    for _ in range(N):
        A.append(int(sys.stdin.readline().strip()))
    
    B = []
    for _ in range(N):
        B.append(int(sys.stdin.readline().strip()))


    result = maxprofit(A,B,N,K)
    print(result)

File: new.cpp
#include<bits/stdc++.h>
using namespace std;


int maxprofit(vector<int> A, vector<int> B, int n, int k)
{
    sort(A.begin(),A.end());

    int i=0,j=1;
    int profit = 0;
    // Ai + Aj <= k;
    int maxp = 0;
    while(i<=j)
    {
        if(j==n)
        {
            i++;
            j=i+1;
        }
        if(A[i] + A[j] <= k && j<n)
        {
            profit += B[i] + B[j];
            j++;
        }
        else
        {
            break;
        }
        maxp = max(profit,maxp);
        profit = 0;
    }
    return maxp;
}


int main()
{
    vector<int> A;
    vector<int> B;

    int n,k;
    cin >> n >> k;

    int i,j,l;

    for(i=0;i<n;i++)
    {
        cin >> l;
        A.push_back(l);
    }
    for(i=0;i<n;i++)
    {
        cin >> l;
        B.push_back(l);
    }

    int result = maxprofit(A,B,n,k);
    cout << result << endl;






    return 0;
}